pkg/controller/machineset/machineset_controller.go

```diff
@@ -230,12 +245,188 @@ func (c *Controller) processNextWorkItem() bool {
 	}
 	defer c.queue.Done(key)
 
+	// create/delete ansible jobs
 	err := c.syncHandler(key.(string))
 	c.handleErr(err, key)
 
+	// create/delete Machine objects
+	err = c.syncHandler2(key.(string))
+	if err != nil {
+		c.logger.Errorf("JDIAZ error handing machine objects")
+	}
+
 	return true
 }
 
+func (c *Controller) syncMachineSet(key string) error {
+	c.logger.WithField("key", key).Debugf("JDIAZ synching machineset %v", key)
+
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		c.logger.Errorf("JDIAZ error parsing namespace out of machineset name %v", key)
+		return err
+	}
+
+	ms, err := c.machineSetsLister.MachineSets(namespace).Get(name)
+	if errors.IsNotFound(err) {
+		c.logger.WithField("key", key).Errorf("JDIAZ couldn't find machineset %v", key)
+		return err
+	}
+	if err != nil {
+		c.logger.Errorf("JDIAZ problem fetching machineset %v", key)
+		return err
+	}
+	if ms.Spec.NodeType == "Master" {
+		// masters aren't cattle, they need ansible depoyment
+		c.logger.WithField("machineset", ms).Debugf("JDIAZ ignoring master machineset: %v", ms)
+		return nil
+	}
+	//c.logger.WithField("machineset", ms).Debugf("JDIAZ full machineset: %+v", ms)
+
+	// check DeletionTimestamp???
+
+	// get all the machines in our current namespace
+	allMachines, err := c.machineLister.Machines(namespace).List(labels.Everything())
+	if err != nil {
+		c.logger.WithField("machineset", ms).Errorf("JDIAZ error while trying to get list of all machines")
+		return err
+	}
+	c.logger.WithField("machineset", name).Debugf("JDIAZ machinelist: %v", allMachines)
+
+	// get all machines which are children of this machineset
+	var filteredMachines []*clusteroperator.Machine
+	for _, machine := range allMachines {
+		if machine.DeletionTimestamp != nil {
+			continue
+		}
+		controllerRef := metav1.GetControllerOf(machine)
+		if controllerRef == nil {
+			c.logger.WithField("machineset", name).Debugf("JDIAZ no controller ref for machine %v", machine)
+			continue
+		}
+		if ms.UID != controllerRef.UID {
+			//c.logger.WithField("machineset", name).Debug("JDIAZ the kid is not my son %v", machine)
+			continue
+		}
+		filteredMachines = append(filteredMachines, machine)
+	}
+	//c.logger.WithField("machineset", name).Debugf("JDIAZ children machines: %v", filteredMachines)
+
+	var manageMachinesErr error
+
+	//if machineSetNeedsSync && ms.DeletionTimestamp == nil {
+	if true && ms.DeletionTimestamp == nil {
+		manageMachinesErr = c.manageMachines(filteredMachines, ms)
+	}
+
```

Comment:
We'll want to add a status update to the machineset to track how many machines currently exist.